realm/src/main/java/io/realm/Realm.java

```diff
@@ -309,59 +282,49 @@ public static Realm getInstance(Context context, boolean autoRefresh) {
      */
     @SuppressWarnings("UnusedDeclaration")
     public static Realm getInstance(Context context, String fileName) {
-        if (Looper.myLooper() != null) {
-            return Realm.create(context.getFilesDir(), fileName, null, true);
-        } else {
-            return Realm.create(context.getFilesDir(), fileName, null, false);
-        }
+        return Realm.getInstance(context, fileName, null);
     }
 
     /**
      * Realm static constructor.
      * {link io.realm.close} must be called when you are done using the Realm instance.
      *
-     * <strong>This constructor is now deprecated and will be removed in version 0.76.0.</strong>
+     * It sets auto-refresh on if the current thread has a Looper, off otherwise.
      *
-     * @param context  an Android context
-     * @param fileName the name of the file to save the Realm to
-     * @param autoRefresh whether the Realm object and its derived objects (RealmResults and RealmObjects)
-     *                    should be automatically refreshed with the event loop (requires to be in a thread with a Looper)
+     * @param context an Android {@link android.content.Context}
+     * @param key     a 64-byte encryption key
      * @return an instance of the Realm class
      * @throws RealmMigrationNeededException The model classes have been changed and the Realm
      *                                       must be migrated
      * @throws RealmIOException              Error when accessing underlying file
-     * @throws java.lang.IllegalStateException The Realm is being instantiated with auto-refresh
-     *                                         in a Thread without a {@link android.os.Looper}
+     * @throws java.lang.IllegalStateException The Realm is being instantiated in a Thread without
```

Comment:
Isn't that wrong now. It only happens if you manually call setAutorefresh, right?